homework-submission/homework-week3/react-hyf/react-hyf-client/src/App.js

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import './App.css';
+import PreviousHomework from './PreviousHomework';
+import HomeworkWeek3 from './HomeworkWeek3';
+
+const url = ' https://uinames.com/api/?amount=10';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { data: [] };
+  }
+  componentDidMount() {
+    this.fetchApi()
+      .then(results => this.setState({ data: results }))
+      .catch(err => console.error(err));
+  }
+  async fetchApi() {
+    const data = await fetch(url);
+    const parsedData = await data.json();
+    if (data.status !== 200) {
+      throw Error(parsedData.message);
```

Comment:
if the response status is not a 200, this line won't be read to show the error message, and `parsedData` won't surly includes any kind of an error message to throw.
What you could do is to move the if statement to the line before `parsedData` and have the condition to specify a custom client Error handling.